### PR TITLE
Initial CI Testing

### LINKED ISSUE #38: Implement Unit Tests for existing 3 controls
**Is your feature request related to a problem? Please describe.**
no

**Describe the solution you'd like**
Additional testing for existing controls with the following areas in mind
    - Check module import for definition (component should be defined)
    - Check DOM structure matches implementation in the component implementation.
    - Check overrides from props change the expected behaviors / style of the control
    - Check call backs are invoked at correct times.
    
**Describe alternatives you've considered**
Please describe alternative solutions or features you have considered.

### PR DESCRIPTION
**Pull request by brokecollegekidwithaclothingobsession**
Initial CI Testing 

## Changes

- Added Unit Tests to LoginButton
- Added Unit Tests to RegisterButton
- Added Unit Tests to RegisterForm
- Added CI to GitHub Actions

## API Updates
N/A

### New Features *(required)*
- Added Unit Testing for 3 Base Components
- Added CI 

### Deprecations *(required)*
N/A

### Enhancements *(optional)*
Unit Tests!

## Checklist

- [X] Unit tests
- [X] Documentation

## References
N/A
(optional)

Fixes #38  
Connects #10 

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

**Review by ntindle — APPROVED** [review]
LGTM for CI, @joshLong145 review for tests

File: src/components/Icons.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import React from 'react';
 export const SonrLogoWrapper = (props: any) => {
     return (
         <div>
-            <Logo className="w-10 h-10 mr-2"/>
+            <Logo data-testid="logo-svg" className="w-10 h-10 mr-2"/>
```
Comment (joshLong145) [inline src/components/Icons.tsx:7]: can you add a comment denoting this is a property for unit testing? developers might not know what the purpose of this property

File: src/components/LoginButton/LoginButton.tsx [hunk 2]
```diff
@@ -21,11 +21,12 @@ export function LoginButton(loginButtonProps: loginButtonProps) {
...
-        <button 
-          className={loginButtonProps.styling}
-          onClick={onClickWrapper(loginButtonProps.onLogin, loginButtonProps.onError)}>
-            <SonrLogoWrapper/>
-          <span className='text-md pr-8'>{loginButtonProps.label}</span>
+        <button
+          id='login-button' 
+          className={props.styling}
+          onClick={onClickWrapper(props.onLogin, props.onError)}>
+            {/* <SonrLogoWrapper/> */}
```
Comment (joshLong145) [inline src/components/LoginButton/LoginButton.tsx:28]: Do we need to comment out all controls using the logo or just the login button?

**Review by joshLong145 — CHANGES_REQUESTED** [review]
left a few comments looks good over all.

File: src/components/LoginButton/LoginButton.tsx (hunk 2 above)
Comment (brokecollegekidwithaclothingobsession) [reply src/components/LoginButton/LoginButton.tsx:28]: Both Login & Register Buttons need to be commented out at this moment for all unit tests to work.

**Review by joshLong145 — COMMENTED** [review]
Looks good, we will have to figure out why inline svgs are erroring with jest / jest-dom. But once we have the designs we can revisit this issue

**Review by joshLong145 — APPROVED** [review]

**Pull request merged**